### DISCUSSION TITLE
Recommended environment

### DISCUSSION THREAD

**Discussion by RobinBastiaan** [spatie/laravel-sql-commenter#4] [Q&A] [answered]
Hey Spatie,

Thanks for the package. I have a question on how and when to use it.

Is it recommended to only have this package enabled when developing? Is there any significant performance or security hit when used in production? Or maybe this package is even more helpful in production because some other tooling is not available there?

**Comment by riasvdv** [comment, marked as answer]
The performance impact should be pretty minimal, the most expensive commenter is the File commenter, which is why you can configure the backtrace limit.

We do recommend using this in staging or production, as that's where you'll see the most use of it. When you're developing locally you usually have more tools available to see where the queries come from (something like laravel-debugbar or Ray)
